code/modules/xenoarchaeology/traits/activators.dm

```diff
@@ -0,0 +1,664 @@
+/*
+	Activators
+	These traits cause the xenoartifact to trigger, activate
+
+	* weight - All activators MUST have a weight that is a multiple of 8
+	* conductivity - If an activator should have conductivity, it will be a multiple of 8 too
+*/
+
+/datum/xenoartifact_trait/activator
+	register_targets = FALSE
+	weight = 8
+	conductivity = 0
+	///Do we override the artifact's generic cooldown?
+	var/override_cooldown = FALSE
+
+//Throw custom cooldown logic in here
+/datum/xenoartifact_trait/activator/proc/trigger_artifact(atom/target, type = XENOA_ACTIVATION_CONTACT, force)
+	SIGNAL_HANDLER
+
+	//Trait check - This is different from an anti artifact check and should be done here to avoid activations, this trait is a helper essentially
+	if(target && HAS_TRAIT(target, TRAIT_ARTIFACT_IGNORE))
+		return FALSE
+	//Stop traits that don't register targets activating when we feel them
+	if(parent?.anti_check(target, type))
+		return FALSE
+	parent.register_target(target, force, type)
+	parent.trigger()
+	return TRUE
+
+/datum/xenoartifact_trait/activator/proc/translation_type_a(datum/source, atom/target)
+	SIGNAL_HANDLER
+
+	trigger_artifact(target)
+
+/datum/xenoartifact_trait/activator/proc/translation_type_b(datum/source, atom/item, atom/target)
+	SIGNAL_HANDLER
+
+	if(check_item_safety(item))
+		return
+	trigger_artifact(target)
+
+/datum/xenoartifact_trait/activator/proc/translation_type_c(datum/source, atom/target, atom/item)
+	SIGNAL_HANDLER
+
+	if(check_item_safety(item))
+		return
+	trigger_artifact(target)
+
+/datum/xenoartifact_trait/activator/proc/translation_type_d(datum/source, atom/target)
+	SIGNAL_HANDLER
+
+	var/atom/A = parent?.parent
+	if(!A.density)
+		return
+	trigger_artifact(target)
```

Comment:
This needs an explanation or description about what these actually do and what the difference is between them